backend/src/main/java/atlantbh/restaurants/services/BaseService.java

```diff
@@ -0,0 +1,66 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.exceptions.RepositoryException;
+import atlantbh.restaurants.models.BaseModel;
+import atlantbh.restaurants.models.PaginatedResult;
+import atlantbh.restaurants.models.filters.BaseFilterBuilder;
+import atlantbh.restaurants.repositories.BaseRepositoryImpl;
+import org.hibernate.service.spi.ServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class BaseService<M extends BaseModel<M>, S extends Enum<S>, F extends BaseFilterBuilder<S, F>, R extends BaseRepositoryImpl<M, S, F>> {
+
+    @Autowired
+    protected R repository;
+
+    public PaginatedResult<M> filter(F filterBuilder) {
+        return repository.find(filterBuilder);
+    }
+
+    public M findById(Long id) {
+        return repository.findById(id);
+    }
+
+    public M get(Long id) {
+        M model = repository.findById(id);
+        if (model != null) {
+            return model;
+        }
+        throw new ServiceException("Model with id: " + id + " not found");
+    }
+
+    public M create(M m) throws ServiceException {
+        try {
+            repository.save(m);
+            return m;
+        } catch (RepositoryException e) {
+            throw new ServiceException(e.getMessage());
+        }
+    }
+
+    public M update(Long id, M data) throws ServiceException {
+        try {
+            M model = get(id);
+            if (model == null) {
```

Comment:
why null check? `get` method will throw service exception if there is no model, then this exception will be thrown from update method to controller.. Same for delete

Reply:
Fixed 8c841a20e489a31c66d177c8682339f9d4f19de0